File: fittness-app/parent-module/selfCoach/src/main/java/tudor/work/model/Exercise.java
package tudor.work.model;


import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.*;
import net.minidev.json.annotate.JsonIgnore;

import javax.persistence.*;
import java.util.Set;


@RequiredArgsConstructor
@Data
@Entity
@Builder
@Table(name = "exercise")
@AllArgsConstructor
@EqualsAndHashCode
@ToString
public class Exercise {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;

    @Column(nullable = false)
    private String name;

    @Column(length = 1024)
    private String description;

    private String descriptionUrl;

    private String exerciseImageStartUrl;

    private String exerciseImageEndUrl;

    private String exerciseVideoUrl;

    //private String coverPhotoUrl;

    @ManyToOne
    @JoinColumn(name ="muscle_group_id")
    private MuscleGroup muscleGroup;

//    @ManyToOne
//    @JoinColumn(name = "adder_id")
//    private User adder;

    @ManyToOne
    @JoinColumn(name ="equipment_id")
    private Equipment equipment;


    private Double rating;


    @ManyToOne
    @JoinColumn(name = "difficulty_id")
    private Difficulty difficulty;


    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;

//    @ManyToMany(mappedBy = "exercises")
//    @JsonIgnore
//    private Set<Workout> workouts;


    //this is on true if this exercise is added by a coach;
    private boolean isExerciseExclusive;

    private boolean hasWeight;

    private boolean hasNoReps;

}

File: fittnes_frontend/lib/models/exercise.dart
import 'dart:ffi';
import 'package:flutter/material.dart';

class Exercise {
  final int id;
  final String name;
  final String description;
  final String descriptionUrl;
  final String exerciseImageStartUrl;
  final String exerciseImageEndUrl;
  final String exerciseVideoUrl;
  final String muscleGroup;
  final String equipment;
  final String difficulty;
  final String category;
  final bool exerciseExclusive;
  final double rating;
  final bool hasNoReps;
  final bool hasWeight;

  Exercise(
      {required this.id,
      required this.name,
      this.description = '',
      this.descriptionUrl = '',
      this.exerciseImageStartUrl = '',
      this.exerciseImageEndUrl = '',
      this.exerciseVideoUrl = '',
      required this.muscleGroup,
      required this.equipment,
      required this.difficulty,
      required this.category,
      required this.exerciseExclusive,
      required this.hasNoReps,
      required this.hasWeight,
      this.rating = 0.0});

  factory Exercise.fromJson(Map<String, dynamic> json) {
    return Exercise(
      id: json['id'],
      name: json['name'],
      description: json['description'] ?? '',
      descriptionUrl: json['descriptionUrl'] ?? '',
      exerciseImageStartUrl: json['exerciseImageStartUrl'] ?? '',
      exerciseImageEndUrl: json['exerciseImageEndUrl'] ?? '',
      exerciseVideoUrl: json['exerciseVideoUrl'] ?? '',
      muscleGroup: json['muscleGroup'],
      equipment: json['equipment'],
      difficulty: json['difficulty'],
      category: json['category'],
      exerciseExclusive: json['exerciseExclusive'],
      hasNoReps: json['hasNoReps'],
      hasWeight: json['hasWeight'],
      rating: json['rating'] ?? 0.0,
    );
  }
}
